worker/src/core/loaders/message-worker/util/contact-preference.ts

```diff
@@ -0,0 +1,119 @@
+import axios from 'axios'
+import { ContactChannel, EmailResultRow } from '../interface'
+import config from '@core/config'
+import { map } from 'lodash'
+import CircuitBreaker from 'opossum'
+
+const options = {
+  timeout: 5000, // Trigger failure if phonebook takes longer than 5sec to respond
+  errorThresholdPercentage: 20, // When 20% of requests fail, open the circuit
+  resetTimeout: 30000, // After 30 seconds, half open the circuit and try again.
+}
+
+type ContactPreferenceRequest = {
+  userChannels: ContactChannel[]
+}
+
+async function getContactPrefLinks(request: ContactPreferenceRequest) {
+  const url = `${config.get(
+    'phonebookContactPref.url'
+  )}/api/v1/generate_pref_links`
+  return axios
+    .post(url, JSON.stringify(request), {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-api-key': config.get('phonebookContactPref.apiKey'),
+      },
+    })
+    .then((resp) => resp.data)
+    .catch((err) => {
+      throw err
+    })
+}
+
+const breaker = new CircuitBreaker(getContactPrefLinks, options)
+
+export const getContactPrefLinksForEmail = async (
+  result: EmailResultRow[],
+  campaignId: number,
+  campaignOwnerEmail: string,
+  channel = 'Email'
```

Comment:
beware the inconsistency, channel in postman codebase  as `campaign_type` enum label is `EMAIL`, just making sure that this is intentional

Reply:
Yes it is intentional